### DISCUSSION TITLE
"broken pipe" error when trying to install on Motorola Edge 2024

### DISCUSSION THREAD

**Discussion by Lost-Entrepreneur439** [NightscoutFoundation/xDrip#3637] [Bug reports]
I've tried both the latest stable build and latest nightly, but when attempting to install both, SystemUI crashed. When attempting to install in ADB I get the following error:
```
ella@debian:~$ adb install /home/ella/Downloads/xDrip-plus-20240807-f96d4e5.apk 
Performing Streamed Install
adb: failed to install /home/ella/Downloads/xDrip-plus-20240807-f96d4e5.apk: cmd: Failure calling service package: Broken pipe (32)
```
The phone is rooted and running microG. Not really sure what other debugging details I can provide since the app won't even install. The phone is running Android 14.

**Comment by Navid200** [comment]
This has to be caused by your phone.  Perhaps related to it having been routed.  
I suspect that because it is the first time I am seeing this and the stable release you are experiencing this with has been out since 4 months ago.  
Android 14 is not new.  We have already tested Android 15 and are ready for it.

I cannot prove this is caused by your phone.  
Can you suggest how this could be caused by the xDrip code?

Let's say we find a way to address this by making a change to the xDrip code.  There is no way we can keep the code going forward maintaining this because none of us, as far as I know, uses a rooted phone.  I have no way of testing xDrip with a rooted phone.

**Reply by Lost-Entrepreneur439** [reply]
I don't believe this is a root issue, all other apps install without problems and I tried to install on another phone, my Samsung Galaxy A20, also rooted and running microG, and it had no problem there.

The Edge is a fairly new device (only being released 3 or so months ago), so maybe it has an additional feature that causes installation to break, maybe Motorola does something weird, I don't really know.

I'm happy to provide anything you may need (such as a SystemUI crash log or logcat) and do testing if wanted.

**Reply by Navid200** [reply]
Can I see the logcat please?  
Thanks

**Reply by Navid200** [reply]
Wait!
What is streamed install?

**Reply by Lost-Entrepreneur439** [reply]
> Wait! What is streamed install?

That's just what ADB says whenever you install an app.

**Reply by Navid200** [reply]
OK
I am sorry I jumped the gun.  

Can you just install xDrip on your phone without ADB?  
Would that work?

**Reply by Lost-Entrepreneur439** [reply]
Doing it without ADB also causes a SystemUI crash. Here's the logcat.
[logcat.txt](https://github.com/user-attachments/files/16683781/logcat.txt)

**Reply by Navid200** [reply]
I'm sorry I have to ask you to be patient until someone who knows a lot more than me can have a look.  I will contact them right now.  They are in a different time zone and very likely sleep right now.

**Reply by Navid200** [reply]
Is this the phone?
https://www.gsmarena.com/motorola_edge_(2024)-13111.php

**Reply by Navid200** [reply]
Can you try this if you haven't already?
https://youtu.be/LgVNEe5NIaQ?si=kI0KlrRprcuUx1Pt

**Reply by Lost-Entrepreneur439** [reply]
I've already done that

**Comment by tzachi-dar** [comment]
What android version is the phone?
Looking at the logs it seems related to:
Unknown source permission in split permission: com.google.android.gms.permission.ACTIVITY_RECOGNITION

**Reply by Lost-Entrepreneur439** [reply]
The phone is running Android 14.

**Comment by Navid200** [comment]
I wonder why the first two lines visible in the logcat have a different date than the rest of the lines.

**Comment by jamorham** [comment]
The logs imply that the package manager is crashing due to some incompatibility with micro-g of the ACTIVITY_RECOGNITION permission. SystemUI crashing looks like a bug of the phone. I'm not sure why this would happen and why we haven't heard of it before as I would expect it to affect anything running micro-g.

Broken pipe is likely the package manager also crashing. In all cases neither the package manager or system-ui should crash as these system components should be able to handle APKs they are not compatible with and return an error message rather than break.

xDrip might be using an old mechanism for ACTIVITY_RECOGNITION as it is quite an old feature implementation but based on the crashes happening on the phone I am suspecting the components there rather than anything xDrip is doing.
